configuration/pih/htmlforms/hiv/section-hiv-state.xml

```diff
@@ -156,10 +156,248 @@
+        if (typeof encounterDate !== "undefined" &amp;&amp; encounterDate !== null &amp;&amp; (encounterDate.length > 0)) {
+            currentEncounterDate = new Date(+encounterDate);
+        }
+
+        let treatmentLineConceptUuid="4dfefa8d-9940-4ad5-800d-f16579b37d54";
+
+         let lastRecordedTreatmenLine = {
+            treatmenLine: "",
+            date: null,
+            encounterDisplay: ""
+        };
+
+        
+           jq.getJSON(apiBaseUrl + "/obs", {
+            concept: treatmentLineConceptUuid,
+            patient: patientUuid,
+            v: 'custom:(uuid,display,obsDatetime,value,encounter:(display,encounterDatetime,location:(display),encounterType:(display)))'
+          },
+          function( data ){
+            if (data.results.length > 0) {
+                for (let index = 0; index &lt; data.results.length; ++index) {
+                    let treatmentLineObs = data.results[index];
+                    let treatmentLineObsDate = new Date(treatmentLineObs.obsDatetime);
+                    let tmpTreatmentLineObs = {
+                      treatmenLine: "",
+                      date: null,
+                      encounterDisplay: ""
+                    };
+
+                    if( (currentEncounterDate.getTime() >=  treatmentLineObsDate.getTime()) ||
+                        (currentEncounterDate.toDateString() === treatmentLineObsDate.toDateString())) {
+                            if (typeof treatmentLineObs.value !== "undefined" &amp;&amp; treatmentLineObs.value !== null) {
+                                tmpTreatmentLineObs.treatmenLine = treatmentLineObs.value;
+                                tmpTreatmentLineObs.date = treatmentLineObsDate;
+                                tmpTreatmentLineObs.encounterDisplay = treatmentLineObs.encounter.display;
+                    }
+                    }
+                    if ( (tmpTreatmentLineObs.date != null) &amp;&amp;
+                        ((lastRecordedTreatmenLine.date == null) || (lastRecordedTreatmenLine.date.getTime() &lt; tmpTreatmentLineObs.date.getTime()) ) ) {
+
+                        lastRecordedTreatmenLine.treatmenLine = tmpTreatmentLineObs.treatmenLine;
+                        lastRecordedTreatmenLine.date = tmpTreatmentLineObs.date;
+                        lastRecordedTreatmenLine.encounterDisplay = tmpTreatmentLineObs.encounterDisplay;
+                    }
+                }
+            }
+
+           
+                if (lastRecordedTreatmenLine.date != null) {
+                  
+                   if(lastRecordedTreatmenLine.treatmenLine.display !==null){
+
+                      setValue('Isoniazide.value', 1505);
+                   
+                    jq('#treatmentLine input[type="radio"]').each(function(i,domEl){
+
+                         if(lastRecordedTreatmenLine.treatmenLine.display==="Primary" || lastRecordedTreatmenLine.treatmenLine.display==="Primaire"  &amp;&amp; i==0){
```

Comment:
If we have the uuid, it might be safer here to match on that instead of display name, in case the display name changes?  If we do keep the display, we should double-check this works in French, Kreyol and English... do we use "Primaire" in both the French and Kreyol locales?

Reply:
Agree with @mogoodrich .  We can not use the display name here, as this is not something we can rely upon.  This should be changed to compare by uuid or mapping, using a utility method ideally that we can reuse elsewhere.